parlai/agents/fid/fid.py

```diff
@@ -352,6 +353,14 @@ def get_retrieved_knowledge(self, message):
 
     def _set_query_vec(self, observation: Message) -> Message:
         retrieved_docs = self.get_retrieved_knowledge(observation)
+        if len(retrieved_docs) > self._n_docs:
+            logging.warning(
+                f'Your `get_retrieved_knowledge` method returned {len(retrieved_docs)} Documents, '
+                f'instead of the expected {self._n_docs}. '
+                f'This agent will only use the first {self._n_docs} Documents. '
+                'Consider modifying your implementation of `get_retrieved_knowledge` to avoid unexpected results.'
```

Comment:
can we also say `'or set the --n-docs parameter accordingly'` to inform the user that they can do that too?